examples/helpers/camera_controller.rs

```diff
@@ -171,8 +171,8 @@ fn run_camera_controller(
                 controller.velocity = Vec3::ZERO;
             }
         }
-        let forward = transform.forward();
-        let right = transform.right();
+        let forward = *transform.forward();
+        let right = *transform.right();
         transform.translation += controller.velocity.x * dt * right
             + controller.velocity.y * dt * Vec3::Y
             + controller.velocity.z * dt * forward;
```

Comment:
If we impl `Mul<Direction3d> for f32` or switch around the operand order, these derefs shouldn't be needed either.